File: src/commonTest/kotlin/com/scoreo/application/UpdateGameTypeUseCaseTest.kt
package com.scoreo.application

import com.scoreo.infrastructure.InMemoryGameTypeRepository
import com.scoreo.domain.model.GameType
import com.scoreo.domain.model.TieBreakRule
import com.scoreo.domain.model.WinCondition
import kotlin.test.Test
import kotlin.test.assertEquals

class UpdateGameTypeUseCaseTest {

    @Test
    fun `updates game type in repository`() {
        val repo = InMemoryGameTypeRepository()
        val addUseCase = AddGameTypeUseCase(repo)
        val updateUseCase = UpdateGameTypeUseCase(repo)
        
        val created = addUseCase("Belote", WinCondition.HIGHEST_SCORE)
        val updated = created.copy(name = "Belote Updated")
        updateUseCase(updated)
        
        val result = repo.findById(created.id)
        assertEquals("Belote Updated", result?.name)
    }

    @Test
    fun `updates tie break rule`() {
        val repo = InMemoryGameTypeRepository()
        val addUseCase = AddGameTypeUseCase(repo)
        val updateUseCase = UpdateGameTypeUseCase(repo)
        
        val created = addUseCase("Golf", WinCondition.HIGHEST_SCORE)
        val updated = created.copy(
            tieBreakRule = TieBreakRule.SECONDARY_SCORE,
            tieBreakCondition = WinCondition.LOWEST_SCORE,
            tieBreakLabel = "Handicap",
        )
        updateUseCase(updated)
        
        val result = repo.findById(created.id)
        assertEquals(TieBreakRule.SECONDARY_SCORE, result?.tieBreakRule)
        assertEquals(WinCondition.LOWEST_SCORE, result?.tieBreakCondition)
        assertEquals("Handicap", result?.tieBreakLabel)
    }

    @Test
    fun `preserves game type id`() {
        val repo = InMemoryGameTypeRepository()
        val addUseCase = AddGameTypeUseCase(repo)
        val updateUseCase = UpdateGameTypeUseCase(repo)
        
        val created = addUseCase("Belote", WinCondition.HIGHEST_SCORE)
        val originalId = created.id
        val updated = created.copy(name = "Belote Updated")
        updateUseCase(updated)
        
        val result = repo.findById(originalId)
        assertEquals(originalId, result?.id)
    }
}

File: src/application/updateGameTypeUseCase.test.ts
import { describe, expect, it } from 'vitest'
import { InMemoryGameTypeRepository } from '../infrastructure/testing/inMemoryGameTypeRepository'
import { AddGameTypeUseCase } from './addGameTypeUseCase'
import { UpdateGameTypeUseCase } from './updateGameTypeUseCase'

describe('UpdateGameTypeUseCase', () => {
  it('updates game type in repository', () => {
    const repo = new InMemoryGameTypeRepository()
    const addUseCase = new AddGameTypeUseCase(repo)
    const updateUseCase = new UpdateGameTypeUseCase(repo)

    const created = addUseCase.invoke('Belote', 'HIGHEST_SCORE')
    updateUseCase.invoke({ ...created, name: 'Belote Updated' })

    expect(repo.findById(created.id)?.name).toBe('Belote Updated')
  })

  it('updates tie break rule', () => {
    const repo = new InMemoryGameTypeRepository()
    const addUseCase = new AddGameTypeUseCase(repo)
    const updateUseCase = new UpdateGameTypeUseCase(repo)

    const created = addUseCase.invoke('Golf', 'HIGHEST_SCORE')
    updateUseCase.invoke({
      ...created,
      tieBreakRule: 'SECONDARY_SCORE',
      tieBreakCondition: 'LOWEST_SCORE',
      tieBreakLabel: 'Handicap',
    })

    const result = repo.findById(created.id)
    expect(result?.tieBreakRule).toBe('SECONDARY_SCORE')
    expect(result?.tieBreakCondition).toBe('LOWEST_SCORE')
    expect(result?.tieBreakLabel).toBe('Handicap')
  })

  it('preserves game type id', () => {
    const repo = new InMemoryGameTypeRepository()
    const addUseCase = new AddGameTypeUseCase(repo)
    const updateUseCase = new UpdateGameTypeUseCase(repo)

    const created = addUseCase.invoke('Belote', 'HIGHEST_SCORE')
    const originalId = created.id
    updateUseCase.invoke({ ...created, name: 'Belote Updated' })

    expect(repo.findById(originalId)?.id).toBe(originalId)
  })
})
